vault/mount.go

```diff
@@ -101,9 +101,7 @@ var (
 
 	untunableMounts = []string{
 		mountPathCubbyhole,
-		mountPathSystem,
 		"audit/",
```

Comment:
If we're being thorough... I wonder if we just change the definition of `untunable`? 

`audit/` likely suffers from the same issue: should audit path be audited (probably, yes)? Same probably holds for `mountPathCubbyhole` -- it seems like perhaps we should leave these two paths _in_, but change the usage to disallow certain tunable options?

https://openbao.org/api-docs/system/mounts/#tune-mount-configuration

Of these fields:

 - `audit_non_hmac_request_keys`
 - `audit_non_hmac_response_keys`

both make sense and none of the others do (TTL not used or would be a breaking change, description is system-set, `listing_visibility` likewise, headers don't matter as we already override that, and plugin_version is the system version).

So I think I'd probably prefer the other PR, in `vault/logical_system.go`:

https://github.com/openbao/openbao/blob/cf24f420ae9ac90c592ba6185e5333c700eed788/vault/logical_system.go#L1565-L1571

and only tune these two options, returning early if it was handled (so that other values aren't affected). My 2c!

Reply:
Agree completely, this makes more sense in the context of the issue I'm trying to fix.

Speaking about being thorough though - I would even argue that setting non-hmac audit keys doesn't logically belong to `tune` at all and should be set independently as  `audit_settings` or something along those lines. But that would be a rather breaking change and I don't think such a change would be warranted on its own (it could be entertained if a larger rework of auditing is done in the future for example).

Reply:
Hmmm, yeah, auditing is one of those weird things where its cross-role really, tbh. A separate set of paths might make some sense; perhaps we could introduce a path like `sys/mounts/:path/audit-tune` which just has those two keys? Tuning I do agree, seems to mostly be the scope of a mount-level admin, but auditing seems to more broadly require collaboration between instance/auditing admins and the admins of individual plugins being SMEs and knowing what is safe to HMAC/non-HMAC.

Reply:
agreed but I think it's off-topic in this MR, created a new feature request https://github.com/openbao/openbao/issues/476

Reply:
@jficz Just so I make sure we're on the same page :D 

You're going to update this to make sure only the hmac fields are modified by the actual tuning code and the other fields aren't, when tuning the sys/ and audit/ paths? And then we'll merge this and let the #476 be addressed later, perhaps by someone else?

Reply:
I can try but it's probably not gonna be as simple as this so I can't really guarantee either code quality or an ETA :) give me a few days to familiarize myself with the codebase and I'll see what I can come up with

Reply:
\o hey @jficz! Still interested in contributing this? Got any questions? Happy to talk through anything if you're stuck. :-)

Reply:
hey @cipherboy! I am, but the project was sidelined for a while and I got other priorities atm. But I still want to return to this either in this Q or in Q1/2025. I don't want to block the work tho so if anyone else is interested in working on this, we can close the draft to not block anyone.